include/glow/Graph/Nodes.h

```diff
@@ -140,6 +140,15 @@ class Placeholder : public Storage {
   /// Specifies if associated Tensors should be zeroed when allocated.
   bool allocZero_{false};
 
+  /// Specifies if this is a cacheable placeholder, this means it should be
+  /// re-used between inferences.
+  bool isCacheable_{false};
+
+  /// Current version of the Placeholder, by default this is 0 but for a
+  /// cacheable placeholder the tensor can be updated, this is reflected by an
+  /// incrememted version number.
+  uint64_t versionNumber_{0};
```

Comment:
Not sure if this can be used from multiple threads - but if so, would suggest using an atomic.